packages/cli/src/metrics/index.ts

```diff
@@ -6,66 +6,113 @@ import type express from 'express';
 import promBundle from 'express-prom-bundle';
 import promClient from 'prom-client';
 import semverParse from 'semver/functions/parse';
+import { Service } from 'typedi';
 
 export { promClient };
 
-export function configureMetrics(app: express.Application) {
-	if (!config.getEnv('endpoints.metrics.enable')) {
-		return;
+@Service()
+export class MetricsService {
+	cacheHitsTotal: promClient.Counter<'cache'> | undefined;
+
+	cacheMissesTotal: promClient.Counter<'cache'> | undefined;
+
+	cacheUpdatesTotal: promClient.Counter<'cache'> | undefined;
+
+	configureMetrics(app: express.Application) {
+		if (!config.getEnv('endpoints.metrics.enable')) {
```

Comment:
while we are at it, maybe we should move this check to `Server.ts`, make the imports dynamic. that way for most users we won't be loading these dependencies unless they explicitly enable the metrics endpoint.

Reply:
made import dynamic